docs/pipelines.md

```diff
@@ -75,22 +83,42 @@ A `Pipeline` definition supports the following fields:
+      - [`taskRef`](#adding-tasks-to-the-pipeline) - a reference to a `Task` definition.
+      - [`taskSpec`](#adding-tasks-to-the-pipeline) - a specification of a `Task`.
+      - [`resources`](#specifying-resources-in-pipelinetasks) - Specifies the [`PipelineResource`](resources.md) that
+        a `Task` requires.
+        - [`from`](#using-the-from-field) - Indicates the data for a [`PipelineResource`](resources.md)
           originates from the output of a previous `Task`.
-      - [`runAfter`](#using-the-runafter-parameter) - Indicates that a `Task`
-        should execute after one or more other `Tasks` without output linking.
-      - [`retries`](#using-the-retries-parameter) - Specifies the number of times to retry the
-        execution of a `Task` after a failure. Does not apply to execution cancellations.
+      - [`runAfter`](#using-the-runafter-field) - Indicates that a `Task` should execute after one or more other 
+        `Tasks` without output linking.
+      - [`retries`](#using-the-retries-field) - Specifies the number of times to retry the execution of a `Task` after
+        a failure. Does not apply to execution cancellations.
       - [`conditions`](#guard-task-execution-using-conditions) - Specifies `Conditions` that only allow a `Task`
         to execute if they successfully evaluate.
+      - [`when`](#guard-finally-task-execution-using-when-expressions) - Specifies `when` expressions that guard
+        the execution of a `Task`; allow execution only when all `when` expressions evaluate to true.
       - [`timeout`](#configuring-the-failure-timeout) - Specifies the timeout before a `Task` fails.
-  - [`results`](#configuring-execution-results-at-the-pipeline-level) - Specifies the location to which
-    the `Pipeline` emits its execution results.
+      - [`params`](#specifying-parameters-in-pipelinetasks) - Specifies the `Parameters` that a `Task` requires.
+      - [`workspaces`](#specifying-workspaces-in-pipelinetasks) - Specifies the `Workspaces` that a `Task` requires.
+  - [`results`](#emitting-results-from-a-pipeline) - Specifies the location to which the `Pipeline` emits its execution
+    results.
   - [`description`](#adding-a-description) - Holds an informative description of the `Pipeline` object.
-  - [`finally`](#adding-finally-to-the-pipeline) - Specifies one or more `Tasks`
-    to be executed in parallel after all other tasks have completed.
+  - [`finally`](#adding-finally-to-the-pipeline) - Specifies one or more `Tasks` to be executed in parallel after 
+    all other tasks have completed.
+    - [`name`](#adding-finally-to-the-pipeline) - the name of this `Task` within the context of this `Pipeline`.
+    - [`taskRef`](#adding-finally-to-the-pipeline) - a reference to a `Task` definition.
+    - [`taskSpec`](#adding-finally-to-the-pipeline) - a specification of a `Task`.
+    - [`retries`](#using-the-retries-field) - Specifies the number of times to retry the execution of a `Task` after
```

Comment:
@vdemeester, here is the answer to your question 🤣 `finally` does support `retries` 😄

Reply:
haha yea, had to test it out to prove we support retries, was confused given our discussion yesterday at pipeline wg 😀